### PR TITLE
matched up alternators with engines

### LINKED ISSUE #16664: Alternator Type Frequency / Possible Mismatches
Just found it odd that car alternators are found in flatbed trucks, pickup trucks, and even an ambulance with a V8 Engine. Seems like some trucks and other larger vehicles should have truck alternators. These are just the ones I have checked so far, but there may be other examples.

### PR DESCRIPTION
**Pull request by Xpyder**
matched up alternators with engines
also updated batteries for affected vehicles

Exceptions:
- Pickup and Flatbed trucks: Assigned truck parts despite gas v6 engines
- Supercar left with car alternator, assuming prioritizing engine power
  over battery charge
- Superbike gave car alternator and car battery, assuming it's basically
  a car engine on a bike frame, assuming car battery needed to start engine
# 

Fixes #16664

### CODE REVIEW

File: data/json/vehicles/emergency.json [hunk 1]
```diff
@@ -50,8 +50,8 @@
             {"x": 1, "y": 2, "part": "windshield"},
             {"x": 2, "y": 0, "part": "frame_horizontal"},
             {"x": 2, "y": 0, "part": "engine_v8"},
-            {"x": 2, "y": 0, "part": "alternator_car"},
-            {"x": 2, "y": 0, "part": "battery_car"},
+            {"x": 2, "y": 0, "part": "alternator_truck"},
+            {"x": 2, "y": 0, "part": "battery_truck"},
```
Comment (Cyrano7) [inline data/json/vehicles/emergency.json]: Why should a police car have a truck battery?

**Comment by illi-kun** [comment]
I'm not sure about superbike related changes (car battery especially, do you think a bike has enough space for that guy?). The rest is fine for me.

**Comment by karols146** [comment]
"Car engine on a bike frame" - sounds weird.

**Comment by Xpyder** [comment]
It's actually using the same inline v4 engine like you'd find in a small car.

**Comment by mugling** [comment]
Whilst your at it the excavator is missing an alternator entirely

**Comment by Xpyder** [comment]
I haven't found anything definitive on bikes using an inline v4, but I do see talks about the cold cranking amps and special batteries for them.  I'm going to set them to use a car battery but a motorbike alternator unless anyone has any objections

File: data/json/vehicles/utility.json [hunk 2]
```diff
@@ -231,6 +231,7 @@
             { "x":  0, "y":  0, "part": "controls" },
             { "x":  0, "y":  0, "part": "horn_big" },
             { "x":  1, "y":  0, "part": "diesel_engine_i6" },
+			{ "x":  1, "y":  0, "part": "alternator_truck"},
```
Comment (mugling) [inline data/json/vehicles/utility.json]: spacing?

Comment (Xpyder) [inline data/json/vehicles/utility.json]: Thanks, I can't fix in web browser so I'll submit in a little bit.  I want to go through all the engines and make sure no others are missing an alternator or battery

**Comment by Xpyder** [comment]
I've also audited all the car parts with the word Engine and verified they all have a battery and (if not electric) an alternator

**Pull request merged**